File: Miscellaneous/Python/1. Two Sum.py
nums = [3,2,4]
target = 6

class Solution:
    def twoSum(self, nums, target):
        result = []
        for i in range(len(nums)-1):
            for j in range(i+1,len(nums)):
                if nums[i] + nums[j] == target:
                   result.append(i)
                   result.append(j)
                return result

    def twoSum2(self, nums, target):
        tempMap = {}
        for i in range(len(nums)):
            if target - nums[i] in tempMap:
                return [tempMap[target-nums[i]], i]
            else:
                tempMap[nums[i]]=i


x = Solution()
print(x.twoSum2(nums, target))
        

File: Miscellaneous/Golang/1. Two Sum.go
package main

import "fmt"

func main() {

	nums := []int{2, 7, 11, 15}
	target := 9

	result := twoSum(nums, target)
	fmt.Println(result)
}

func twoSum(nums []int, target int) []int {
	var ans []int
	tempMap := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		_, ok := tempMap[target-nums[i]]
		if ok {
			ans = append(ans, tempMap[target-nums[i]])
			ans = append(ans, i)
			break
		}
		tempMap[nums[i]] = i
	}

	return ans
}
